Use the configured COPR project when triggering Testing Farm

csomh opened the pull request:
Before this, we always referred to the default COPR project when
triggering Testing Farm. This caused tests to fail in repos which choose
to build in their own COPR namespace and project.

Relates to #523.

Signed-off-by: Hunor Csomortáni <csomh@redhat.com>

csomh commented:
Okay, so this took a little longer and ended up also making a few tweaks for how to run tests locally, this is why there are 7 (seven) commits.

The first three are related to the subject of this PR.
The following three are a few Makefile tweaks - I can brake this out in a separate PR, if you say so.
The last is to get rid of some pytest warnings - I can brake this out in a separate PR, if you say so.

See individual commit messages for the motivation of these changes.

TomasTomecek commented on packit_service/worker/testing_farm.py:
```diff
@@ -116,36 +137,8 @@ def run_testing_farm(self, chroot: str) -> HandlerResults:
             chroot=chroot,
         )
 
-        pipeline_id = str(uuid.uuid4())
-        logger.debug(f"Pipeline id: {pipeline_id}")
-
-        test_run_model = TFTTestRunModel.create(
-            pipeline_id=pipeline_id,
-            commit_sha=self.event.commit_sha,
-            status=TestingFarmResult.new,
-            target=chroot,
-            web_url=None,
-            trigger_model=self.event.db_trigger,
-        )
```
where did this code go?

csomh replied:
facepalm

lachmanfrantisek commented on tests/unit/test_testing_farm.py:
```diff
@@ -20,31 +20,35 @@
+# These names are definately not nice, still they help with making classes
+# whose names start with Testing* or Test* to become invisible for pytest,
+# and so stop the test discovery warnings.
+from packit_service.service.events import TestingFarmResultsEvent as TFResultsEvent
```
Good catch. You can rename the actual class.

TomasTomecek replied:
I'd rename it to TFT :)

csomh replied:
I actually like the current name of the class: it's much easier to read and understand than TF-something.

csomh replied:
Sorry, no, not this time, no more renaming :)

These aliases are local to this test, if you really think renaming these classes is useful then lets do this in some other PR. But as I said above, the current names are perfectly fine from a readability point of view.

csomh commented:
recheck

lachmanfrantisek commented:
Probably some flexmocking problem:

```
[1m    def checkout_ref(self, ref: str):[0m
[1m        """ Check out selected ref in the git repo; equiv of git checkout -B ref """[0m
[1m        logger.info(f"Checking out ref {ref}.")[0m
[1m>       if ref not in self.git_repo.branches:[0m
[1m[31mE       AttributeError: 'NoneType' object has no attribute 'branches'[0m
```

csomh commented:
> Probably some flexmocking problem:
> 

Yeah, I spent the last 2 hours looking into this, but I'm still not able to tell what's the issue, and I'm unable to reproduce this locally. (I've updated/rebuilt all my local images, made sure everything is freshly pulled.)

lachmanfrantisek commented:
@csomh I'll take a look if you want, it looks familiar to me.

csomh commented:
> @csomh I'll take a look if you want, it looks familiar to me.

That would be sooooo great! :hugs:

I've updated this to only include the fix commit, and removed all the others. Let's see how the tests are doing. 

The rest of the commits you can pull it from

`https://github.com/csomh/packit-service test-curstom-copr-project`

lachmanfrantisek commented:
> That would be sooooo great! hugs

lachmanfrantisek commented:
Sorry, wrong button.

It passes for me locally as well -- I need to find the right version of dependencies to reproduce it.
